packages/keychain/src/components/Funding.tsx

```diff
@@ -0,0 +1,390 @@
+  const [tokens, setTokens] = useState<TokenInfo[]>([]);
+
+  const remaining = useMemo(() => tokens.filter((t) => !t.isFunded), [tokens]);
+
+  useEffect(() => {
+    const target = [
+      {
+        address: ETH_CONTRACT,
+        min: "10000000000000000",
+      },
+      ...prefunds,
+    ];
+    if (!target.length) return;
+    fetchTokneInfo();
+
+    async function fetchTokneInfo() {
+      const res = await fetch("https://mainnet-api.ekubo.org/tokens");
+      const data: TokenInfoRaw[] = await res.json();
+      const tokens = target.map((t) => {
+        const info = data.find(
+          ({ l2_token_address }) => l2_token_address === t.address,
+        );
+
+        return {
+          address: t.address,
+          min: t.min,
+          name: info.name,
+          symbol: info.symbol,
+          decimals: info.decimals,
+          logo:
+            t.address === ETH_CONTRACT ? (
+              <EthereumIcon fontSize={20} />
+            ) : (
+              <Image
+                src={info.logo_url}
+                alt={`${info.name} ERC-20 Token Logo`}
+                h={5}
+              />
+            ),
+          isFunded: false,
+        };
+      });
+      setTokens(tokens);
+    }
+  }, [prefunds, controller.account.address]);
+
+  const setIsFundedBulk = useCallback(
+    (funded: TokenInfo[]) => {
+      if (funded.length === 0) return;
+
+      const fundedAddrs = funded.map((f) => f.address);
+      const newTokens = tokens.map((t) => ({
+        ...t,
+        isFunded: fundedAddrs.includes(t.address) ? true : t.isFunded,
+      }));
+
+      setTokens(newTokens);
+    },
+    [tokens],
+  );
+
+  const checkFunds = useCallback(async () => {
+    const funded = await Promise.allSettled(
+      remaining.map(async (t) => {
+        const { abi } = await controller.account.getClassAt(t.address);
+        const contract = new Contract(abi, t.address, controller.account);
+        const { balance } = await contract.balanceOf(t.address);
+
+        return {
+          ...t,
+          isFunded: uint256.uint256ToBN(balance) >= BigInt(t.min),
+        };
+      }),
+    );
+    const res = funded
+      .filter((res) => res.status === "fulfilled" && res.value.isFunded)
+      .map((res) => (res as PromiseFulfilledResult<TokenInfo>).value);
+
+    if (res.length) {
+      setIsFundedBulk(res);
+    }
+  }, [remaining, setIsFundedBulk, controller.account]);
+
+  useInterval(checkFunds, 3000);
+
+  return { tokens, remaining, isAllFunded: !remaining };
+}
+
+const ETH_CONTRACT =
+  "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7";
+
+type TokenInfoRaw = {
+  name: string;
+  symbol: string;
+  decimals: number;
+  l2_token_address: string;
+  sort_order: string;
+  total_supply: number;
+  logo_url: string;
+};
+
+type TokenInfo = {
+  name: string;
+  symbol: string;
+  decimals: number;
+  address: string;
+  logo: React.ReactNode;
+  min: string;
+  isFunded: boolean;
+};
```

Comment:
- `TokenInfo` should have `balance: bigint` field.
- Should I convert `min` to bigint
- `isFunded` can be computed by `t.balance >= t.min`.